aspnetcore/signalr/background-services.md

```diff
@@ -0,0 +1,78 @@
+---
+title: SignalR with Background Services
+author: bradygaster
+description: Learn how to send messages to SignalR clients from .NET Core BackgroundService classes.
+monikerRange: '>= aspnetcore-2.2'
+ms.author: bradyg
+ms.custom: mvc
+ms.date: 01/31/2018
+uid: signalr/background-services
+---
+# Hosting SignalR in Background Services
+
+By [Brady Gaster](https://twitter.com/bradygaster)
+
+This article provides guidance for hosting SignalR Hubs using background worker process hosted with ASP.NET Core, and how to send messages to connected clients from within a .NET Core [BackgroundService](/dotnet/api/microsoft.extensions.hosting.backgroundservice).
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/signalr/background-services/sample/) [(how to download)](xref:index#how-to-download-a-sample)
+
+## Wiring up SignalR during Startup
+
+Hosting ASP.NET Core SignalR Hubs in the context of a background worker process is identical to that of hosting a Hub in an ASP.NET Core web app. In the `ConfigureServices` method of a `Startup` class, calling `services.AddSignalR` adds the required services to the ASP.NET Core DI layer to support SignalR. Then, in `Configure`, the `UseSignalR` method is called to wire up the Hub endpoint(s) in the ASP.NET Core request pipeline.
+
+[!code-csharp[Startup](background-service/sample/Server/Startup.cs?name=Startup)]
+
+In the example code, the `ClockHub` class implements the `Hub<T>` class to create a strongly-typed Hub. The `ClockHub` has been configured in the `Startup` class to respond to requests at the endpoint `/hubs/clock`.
```

Comment:
```suggestion
In the preceding example, the `ClockHub` class implements the `Hub<T>` class to create a strongly typed Hub. The `ClockHub` has been configured in the `Startup` class to respond to requests at the endpoint `/hubs/clock`.
```

Reply:
i tried to avoid terms like "preceding" and "below" per feedback i've consistently received from @tdykstra and switched to using "example code" in most cases. Is there a convention for referring to the example code?